src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs

```diff
@@ -85,21 +97,154 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
+                }
+                else if (!textFormatArgs.HasDateTimeFmt && formatStr[i] == ',' && !hasNumericCharacters)
+                {
+                    // If there is no numeric format character before group separator character, then treat it as an escaping character.
+                    formatStr = formatStr.Insert(i, "\\");
+                    i++;
+                }
+                else if (!textFormatArgs.HasDateTimeFmt && formatStr[i] == '.')
+                {
+                    // Reset hasNumericCharacters to false to later check if any numeric character after decimal point.
+                    decimalPointIndex = i;
+                    hasNumericCharacters = false;
+                }
+                else if (_unsupportedCharacters.Contains(formatStr[i]))
+                {
+                    // Block all unsupported characters
+                    return false;
+                }
+                else if (formatStr[i] == ':')
+                {
+                    hasColonWithNum = true;
+                }
+                else if (formatStr[i] == ';')
+                {
+                    // Does not allow number of section are more than 2.
+                    sectionCount++;
+                    if (sectionCount > 2)
+                    {
+                        return false;
+                    }
+                }
+                else if (i == formatStr.Length - 1)
+                {
+                    // If format string ends with backsplash but no following character (both numeric and datetime format), format is invalid.
+                    if (formatStr[i] == '\\')
+                    {
+                        return false;
+                    }
+
+                    // If format string ends with e or e+ (not escaping character) then format is invalid (only applied for numeric format)
+                    if (textFormatArgs.HasNumericFmt & (formatStr[i] == 'e' || (i > 2 && formatStr[i - 2] != '\\' && formatStr[i - 1] == 'e' && formatStr[i] == '+')))
+                    {
+                        return false;
+                    }
+                }
+                else if (formatStr[i] == '\\')
+                {
+                    if (textFormatArgs.HasNumericFmt || (i < formatStr.Length - 1 && formatStr[i + 1] == '\"'))
+                    {
+                        // Skip next character if seeing escaping character \.
+                        i++;
+                    }
+                    else
+                    {
+                        // Update \c to "c" to match with                       
+                        formatStr = formatStr.Insert(i + 2, "\"");
+                        formatStr = formatStr.Insert(i + 1, "\"");
+                        formatStr = formatStr.Remove(i, 1);
+                        i += 2;
+                    }
+                }
+                else if (formatStr[i] == '\"' && i < formatStr.Length - 1)
```

Comment:
>formatStr[i] == '\"' && i < formatStr.Length -  [](http://example.com/codeflow?start=25&length=47)

If we see a `"` and this is the last character then we should fail. Does it? A test such as `Text(123.456, "0.0""")` should catch that.

Reply:
Add this check and add tests.